src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -107,8 +104,8 @@ protected function processValue($value, $isRoot = false)
             $autowiredMethods = $this->getMethodsToAutowire($reflectionClass);
             $methodCalls = $value->getMethodCalls();
 
-            if ($constructor = $reflectionClass->getConstructor()) {
-                array_unshift($methodCalls, array($constructor->name, $value->getArguments()));
+            if ($constructor = $this->getConstructor($value, false)) {
+                array_unshift($methodCalls, array($constructor, $value->getArguments()));
```

Comment:
Can't we keep `->name` to remove the dedicated case, line 183 (https://github.com/symfony/symfony/pull/22277/files#diff-62df969ae028c559d33ffd256de1ac49R183)?